### DISCUSSION TITLE
Logisim turns downloaded files into .circ files?

### DISCUSSION THREAD

**Discussion by Abod-Shashtari** [logisim-evolution/logisim-evolution#1449] [Q&A] [answered]
Since I have installed Logisim on my pc when I download a file like (.mp4, .mkv) they will be renamed to be .circ files does any one know how can I fix it (how can I stop that)? btw there is someone like me has the same problem and he post this question  on stack overflow and here is the link : https://stackoverflow.com/questions/72023499/logisim-turns-downloaded-files-into-circ-files .

**Comment by MarcinOrlowski** [comment]
FYI LS-e is not involved in download process in your browser. Just out of curiosity - tried uninstalling LS-e to see if that changes anything?

**Reply by Abod-Shashtari** [reply]
I have tried this and you can see my reply down below

**Comment by maehne** [comment]
I cannot reproduce the described behaviour neither got comparable reports from our students, who use Logisim-evolution regularly in our first year courses. Logisim-evolution only registers `.circ` files with the OS to allow their direct opening in the correct software. So, I have no idea how it could interfere with your downloads. Only if you can provide us with more details on your setup and detailed steps how to reproduce the issue, we may be able to help you. As @MarcinOrlowski suggests, you should try uninstalling Logisim-evolution to see if it fixes the problem. Also, you may check with a different browser if it shows the same behaviour. You may installed some browser add-on, which causes the observed behaviour or some small tool, which runs in the background, which interferes with your downloads.

**Reply by Abod-Shashtari** [reply]
I have tried to uninstall Logisim and the problem solved .mp4 files are no converted to .circ so after that I tried to install it again the problem came back again. Also I have tried different browser and again the same problem but if I used a download manger every thing will be ok .mp4 will stay .mp4. And as I mentioned I am not alone facing this problem there is someone on internet have this problem and you can check the Stackoverflow link above.
Note : I am using windows 11 , I have tried google chrome + microsoft edge.

**Reply by Abod-Shashtari** [reply]
I found this on reddit : https://www.reddit.com/r/WindowsHelp/comments/pda4xz/file_downloaded_as_circ/

**Reply by MarcinOrlowski** [reply]
If you feel this is a bug on our side (which is always a chance), please fill a ticket, but please be aware that due to limited bandwidth we will not be able to investigate and because we feel it to be pretty unlikely our fault (but again, I may be wrong or i.e. we may have trigger this issue somehow) it will be a really low priority for us. So to sum things up: we can fix it if we are proven guilty. We'll not be investigating really as there are other works in the queue that we prioritize higher. If you OK with that and really want to help, please fill a [ticket](https://github.com/logisim-evolution/logisim-evolution/issues), then investigate and let us know.

**Comment by MarcinOrlowski** [comment]
Just for a record - similar report:

* #1471

**Comment by aaronsingline** [comment]
I also have this bug and quite frankly it's not something that LSe should be doing at all.  Stepping outside its bounds and impacting browser downloads is quite the bug imho.

I'm on Windows 10 21H2, LSe 3.7.2 installed via the .msi file, Chrome 104.0.5112.81.

Happy to help out with any diagnostic data/trouble shooting.

**Reply by maehne** [reply]
It would be very helpful if you could provide URL(s), on which you observe this behaviour. We haven‘t been able to reproduce it on our side, which makes it very hard for us to identify the cause of the problem.

**Reply by maehne** [reply]
@aaronsingline: Please check issue #1471 and report us any additional findings/URLs, which may help us to reproduce the issue.

**Comment by DenB22** [comment]
Same issue as OP. Well we noticed the issue with long named .exe files or possibly long names with dashes and dots - specially Dell Drivers. (short, simple filename .exe appear to be ok)
It happens for the Edge Browser and Chrome. Firefox seems ok, well for the .exe issue. I have not checked .mp4.
I tried the softly and targeted approach with registry changes etc 

Finally a brute force attack has seemed to work, which probably broke loads of other things.
I uninstalled LogSim Evolution, installed CCleaner and ran a registry check and accepted all 2,150 changes! 
(just to be verbose, uninstalling LogiSim without CCleaner did not fix the issue)

I am going to try again on another system and this time just delete .circ (or all mentions of .circ) in the registry after uninstalling LogiSim, but we still want to use LogiSim Evolution and cannot. 
So still not a solution for us really.

Edition	Windows 10 Enterprise
Version	21H2
Installed on	‎20/‎08/‎2020
OS build	19044.1889
Experience	Windows Feature Experience Pack 120.2212.4180.0

Latest versions of Chrome and Edge. LogiSim v3.7.2 (just tried the msi installer for v3.5.0 - same issues)
Just in case - Logisim-evolution.jar v 2.13.8 does not cause any problems.

**Reply by maehne** [reply]
@DenB22: Please, add your findings to issue #1471. Is the issue limited to the provided MSI installer or can it be reproduced when downloading and executing only the JAR file of the current Logisim-evolution version as well? We still cannot reproduce the issue on our side, as none of the reporters was able to provide URLs to websites, which provide downloads that trigger the issue. This makes it difficult for us to investigate the issue.

**Comment by Muph0** [comment]
I have just reproduced this. I was downloading an NVIDIA Nsight .msi installer from [this link](https://developer.nvidia.com/downloads/assets/tools/secure/nsight-graphics/2025_3_0/windows/NVIDIA_Nsight_Graphics_2025.3.0.25148.msi)

I've been a long time Logisim-evolution user and this has happened for the first time.
Logisim v3.8.0

Edition	Windows 11 Pro
Version	24H2
Installed on	‎22.‎10.‎2024
OS build	26100.4652
Experience	Windows Feature Experience Pack 1000.26100.128.0

<img width="1515" height="647" alt="image" src="https://github.com/user-attachments/assets/099f0fcf-fb47-4c1f-b1b0-0f3ad8b28e7c" />

**Reply by Abod-Shashtari** [reply, marked as answer]
I think they solved the issue in version 3.9.0.
just uninstall your current version and install the new one, this should fix the problem.

**Reply by maehne** [reply]
The corresponding issue #1471 was fixed by PR #1765. Please use the [latest release](https://github.com/logisim-evolution/logisim-evolution/releases) of [current Nightly build](https://github.com/logisim-evolution/logisim-evolution?tab=readme-ov-file#nightly-builds), which contains a lot of bug fixes and improvements. I will close this discussion as resolved. If the problem persists in the Nightly build, please create a new issue and refer to issue #1471.
